usr/lib/common/encr_mgr.c

```diff
@@ -523,13 +525,23 @@ CK_RV encr_mgr_init(STDLL_TokData_t *tokdata,
         memset(ctx->context, 0x0, sizeof(AES_CONTEXT));
         break;
     case CKM_AES_GCM:
-        if (mech->ulParameterLen != sizeof(CK_GCM_PARAMS) ||
+        if ((mech->ulParameterLen != sizeof(CK_GCM_PARAMS) &&
+             mech->ulParameterLen != sizeof(CK_GCM_PARAMS_COMPAT)) ||
             mech->pParameter == NULL) {
             TRACE_ERROR("%s\n", ock_err(ERR_MECHANISM_PARAM_INVALID));
             rc = CKR_MECHANISM_PARAM_INVALID;
             goto done;
         }
 
+        if (mech->ulParameterLen == sizeof(CK_GCM_PARAMS_COMPAT)) {
+            aes_gcm_param_from_compat((CK_GCM_PARAMS_COMPAT *)mech->pParameter,
+                                      &gcm_params);
```

Comment:
Routine aes_gcm_param_from_compat has a return code CK_RV, which is CKR_ARGUMENTS_BAD if one or both of the parms are NULL. Here, the routine is used without rc checking. However, it's obvious that the parms are not NULL. Just to mention ...

Reply:
will change that function to void. Its an internal function, and the parameters have been NULL checked already.